phenol-annotations/src/main/java/org/monarchinitiative/phenol/annotations/io/hpo/HpoaDiseaseDataLoaderDefault.java

```diff
@@ -36,30 +34,40 @@ public HpoaDiseaseDataContainer loadDiseaseData(InputStream is) throws IOExcepti
 
     Map<TermId, List<HpoAnnotationLine>> annotationLinesByDiseaseId = new HashMap<>();
     boolean header = false;
-    HpoAnnotationMetadata metadata = new HpoAnnotationMetadata();
-    StringBuilder metaDataBuilder = new StringBuilder();
+    String version = null;
+    String hpoVersion = null;
     for (String line = reader.readLine(); line != null; line = reader.readLine()) {
       // Comment Lines
       if (line.startsWith("#")) {
-        metaDataBuilder.append(String.format("%s\n", line.replace("#", "")));
+        if(line.startsWith("#version:") || line.startsWith("#date:")){
+          Matcher matcher = VERSION_PATTERN.matcher(line);
+          if (matcher.matches()){
+            version = matcher.group("version");
+          }
+        } else if(line.toLowerCase().startsWith("#hpo-version:")){
+            Matcher matcher = HPO_VERSION_PATTERN.matcher(line);
+            if (matcher.matches()){
+              hpoVersion = matcher.group("hpoversion");
+            }
+        }
         continue;
       }
 
       // Header Line
-      // Build our yml for metadata to HpoaMetadata
       if (!header) {
-        Constructor c = new Constructor(HpoAnnotationMetadata.class, new LoaderOptions());
-        c.setPropertyUtils(new PropertyUtils() {
-          @Override
-          public Property getProperty(Class<? extends Object> type, String name){
-            name = CaseUtils.toCamelCase(String.join(" ", name.split("\\-")), false);
-            return super.getProperty(type, name);
-          }
-        });
-        final Yaml yaml = new Yaml(c);
-        metadata = yaml.load(metaDataBuilder.toString());
+        if (version == null || (!version.equals(hpoVersion))){
```

Comment:
Hi @iimpulse I think this is overly restrictive. We want to allow working with more recent HPO releases since they are backwards compatible. 

There are two things that can happen in a new release:
- adding new `Term` -  not an issue for HPOA parsing since the new term will not be present in the HPOA file.
- removing `Term` - the term removal should designate the replacement. The replacement will have the `TermId` of the removed `Term` in alternate IDs. We can resolve the replacement when we encounter the removed term during HPOA processing.

We cannot process HPOA with older HPO because HPOA may contain terms that are not in HPO and that will be hard error. Then we must fail and ask the user to update HPO. However, this is outside of the scope of this class. Note that the loader does not use `Ontology hpo` for parsing.

I'll commit a bunch of changes to relax this.